class.jetpack-gutenberg.php

```diff
@@ -529,6 +534,12 @@ public static function enqueue_block_editor_assets() {
 		$editor_script = plugins_url( "{$blocks_dir}editor{$beta}.js", JETPACK__PLUGIN_FILE );
 		$editor_style  = plugins_url( "{$blocks_dir}editor{$beta}{$rtl}.css", JETPACK__PLUGIN_FILE );
 
+		$editor_deps_path = JETPACK__PLUGIN_DIR . $blocks_dir . "editor{$beta}.deps.json";
+		$editor_deps      = file_exists( $editor_deps_path )
+			? json_decode( file_get_contents( $editor_deps_path ) )
+			: array();
+		$editor_deps[]    = 'wp-polyfill';
```

Comment:
What's the rationale behind hardcoding `wp-polyfill` here instead of including it in the deps lists?

Reply:
`wp-polyfill` is unique in that it's not something that the webpack plugin detect for (at this time).

We could actually require `@wordpress/polyfill` from the scripts or add the dependency as part of the webpack plugin. I'd be inclined to keep this initially because it seems to be the easiest to change in the future if we can find a smarter way of doing this 🙂

Reply:
I might be able to answer that :slightly_smiling_face: The deps lists are auto-generated by a webpack plugin that scans JS code for `@wordpress/` imports et al. Polyfills are a bit of an outlier there since they're not typically imported directly but added e.g. through some babel config.

It's arguable that the webpack plugin's output should truthfully reflect those explicit `import`s, and that it's up to the consumer of the `deps` files to also add stuff that's only implicitly referred to.

Reply:
Sorry Jon, hadn't seen yours since I hadn't reloaded this tab in a while :man_facepalming: